gstreamer/src/tags.rs

```diff
@@ -554,6 +570,179 @@ impl StaticType for TagListRef {
     }
 }
 
+#[cfg(feature = "serde-support")]
+struct TagSer<'a>(&'a str, Value);
+
+#[cfg(feature = "serde-support")]
+impl<'a> Serialize for TagSer<'a> {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let mut tup = serializer.serialize_tuple(2)?;
+        tup.serialize_element(self.0)?;
+        match self.1.type_() {
+            // TODO: use a macro serialize_tag_value!
+            glib::Type::String => tup.serialize_element(&self.1.get::<String>().unwrap())?,
+            glib::Type::U32 => tup.serialize_element(&self.1.get::<u32>().unwrap())?,
+            glib::Type::U64 => tup.serialize_element(&self.1.get::<u64>().unwrap())?,
+            _ => {
+                return Err(
+                    ser::Error::custom(
+                        format!("unimplemented Tag serialization for type {}", self.1.type_())
+                    )
+                );
+            }
+        }
+        tup.end()
+    }
+}
+
+#[cfg(feature = "serde-support")]
+impl Serialize for TagList {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let tag_count = unsafe { ffi::gst_tag_list_n_tags(self.as_ptr()) };
```

Comment:
@ystreet wanted an API to be able to iterate over all tags for reading and having them as `Value`s then. This would simplify this code here a lot

Reply:
Yes, that would help.

Reply:
My current WIP is at https://github.com/sdroege/gstreamer-rs/pull/107 for this :)